src/Messages.jl

```diff
@@ -0,0 +1,555 @@
+ 2xx (Successful) responses to a CONNECT request method (Section 4.3.6 of
+ [RFC7231]) switch to tunnel mode instead of having a message body.
+ All 1xx (Informational), 204 (No Content), and 304 (Not Modified)
+ responses do not include a message body.  All other responses do
+ include a message body, although the body might be of zero length."
+[RFC7230 3.3](https://tools.ietf.org/html/rfc7230#section-3.3)
+"""
+
+bodylength(r::Response)::Int =
+                 r.request.method == "HEAD" ? 0 :
+                     r.status in [204, 304] ? 0 :
+    (l = header(r, "Content-Length")) != "" ? parse(Int, l) :
+                                              unknown_length
+
+
+"""
+"The presence of a message body in a request is signaled by a
+ Content-Length or Transfer-Encoding header field.  Request message
+ framing is independent of method semantics, even if the method does
+ not define any use for a message body."
+[RFC7230 3.3](https://tools.ietf.org/html/rfc7230#section-3.3)
+"""
+
+bodylength(r::Request)::Int =
+    ischunked(r) ? unknown_length :
+                   parse(Int, header(r, "Content-Length", "0"))
+
+
+"""
+    readbody(::IO, ::Parser) -> Vector{UInt8}
+
+Read message body from an `IO` stream.
+"""
+
+function readbody(io::IO, parser::Parser, m::Message)
+    if ischunked(m)
+        body = IOBuffer()
+        while !bodycomplete(parser) && !eof(io)
+            data, excess = parsebody(parser, readavailable(io))
+            write(body, data)
+            unread!(io, excess)
+        end
+        m.body = take!(body)
+    else
+        l = bodylength(m)
+        m.body = read(io, l)
+        if l != unknown_length && length(m.body) < l
+            throw(EOFError())
+        end
+    end
+end
+
+
+function Base.parse(::Type{T}, str::AbstractString) where T <: Message
+    bytes = IOBuffer(str)
+    p = Parser()
+    r = Request()
+    m::T = T == Request ? r : r.response
+    readheaders(bytes, p, m)
+    readbody(bytes, p, m)
+    readtrailers(bytes, p, m)
+    if ischunked(m) && !messagecomplete(p)
+        throw(EOFError())
+    end
+    return m
+end
+
+
+"""
+    set_show_max(x)
+
+Set the maximum number of body bytes to be displayed by `show(::IO, ::Message)`
+"""
+
+set_show_max(x) = global body_show_max = x
+body_show_max = 1000
+
+
+"""
+    bodysummary(bytes)
+
+The first chunk of the Message Body (for display purposes).
+"""
+bodysummary(bytes) = view(bytes, 1:min(length(bytes), body_show_max))
+
+function compactstartline(m::Message)
+    b = IOBuffer()
+    writestartline(b, m)
+    strip(String(take!(b)))
+end
+
+function Base.show(io::IO, m::Message)
+    if get(io, :compact, false)
+        print(io, compactstartline(m))
+        if m isa Response
+            print(io, " <= (", compactstartline(m.request::Request), ")")
+        end
+        return
+    end
+    println(io, typeof(m), ":")
+    println(io, "\"\"\"")
+    writeheaders(io, m)
+    summary = bodysummary(m.body)
```

Comment:
I also found it helpful to detect zip/gzip bodies and not display gzip junk if so. You can test by hitting `http://httpbin.org/gzip`

Reply:
Yeah, that might be a good feature. However, when I started this, I was finding that too much logging cleverness was causing functional bugs (#131). At this point I'd be happier to leave it out for now and add it back when functionality is rock solid. I've updated #141.